packages/@guardian/source-react-components-development-kitchen/src/toggle-switch/ToggleSwitch.tsx

```diff
@@ -83,12 +87,16 @@ export const ToggleSwitch = ({
 	};
 
 	return (
-		<label css={[labelStyles, cssOverrides]} {...props} id="notify">
+		<label
+			css={[labelStyles, cssOverrides]}
+			{...props}
+			id={labelId ?? 'notify'}
```

Comment:
Maybe we can use `generateSourceId` to generate a unique ID for this component if one is not passed.

```tsx
import { generateSourceId } from '@guardian/source-foundations'`

...

<label
			css={[labelStyles, cssOverrides]}
			{...props}
			id={labelId ?? generateSourceId()}
```

Reply:
Or what might be better is if the consumer of this component didn't have to care about the fact that there's a label.

```tsx
import { generateSourceId, descriptionId } from '@guardian/source-foundations'`

const ToggleSwitch = ({ id }) => {
  const buttonId = id || generateSourceId();
  const labelId = descriptionId(buttonId);

  return (
    <label id={labelId}>
      <button id={buttonId} aria-labelledby={labelId} />
   </label>
  )
```